api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -138,6 +140,12 @@
     @Parameter(name = ApiConstants.SSH_KEYPAIR, type = CommandType.STRING, description = "name of the ssh key pair used to login to the virtual machine")
     private String sshKeyPairName;
 
+    @Parameter(name = ApiConstants.POD_ID, type = CommandType.UUID, entityType = PodResponse.class, description = "destination Pod ID to deploy the VM to - parameter available for root admin only")
```

Comment:
Since the feature is defined for root admins, please move the changes to `DeployVMCmdByAdmin` @anuragaw - we don't want normal users to be able to select pod and cluster. Also, see if you want to move the hostId option (added in 923f562aa8d227332477261eb36d4156a1d21308).

Reply:
That's a very good point. Let me investigate and see possibilities.